SU2_CFD/src/solvers/CIncEulerSolver.cpp

```diff
@@ -2337,132 +1857,123 @@ void CIncEulerSolver::BC_Far_Field(CGeometry *geometry, CSolver **solver_contain
   unsigned short iDim;
   unsigned long iVertex, iPoint, Point_Normal;
 
-  su2double *V_infty, *V_domain;
+  const bool implicit = config->GetKind_TimeIntScheme() == EULER_IMPLICIT;
+  const bool viscous = config->GetViscous();
 
-  bool implicit      = config->GetKind_TimeIntScheme() == EULER_IMPLICIT;
-  bool viscous       = config->GetViscous();
-
-  su2double *Normal = new su2double[nDim];
+  su2double Normal[MAXNDIM] = {0.0};
 
   /*--- Loop over all the vertices on this boundary marker ---*/
 
   for (iVertex = 0; iVertex < geometry->nVertex[val_marker]; iVertex++) {
     iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
 
-    /*--- Allocate the value at the infinity ---*/
-
-    V_infty = GetCharacPrimVar(val_marker, iVertex);
-
     /*--- Check if the node belongs to the domain (i.e, not a halo node) ---*/
 
-    if (geometry->nodes->GetDomain(iPoint)) {
-
-      /*--- Index of the closest interior node ---*/
+    if (!geometry->nodes->GetDomain(iPoint)) continue;
 
-      Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+    /*--- Allocate the value at the infinity ---*/
 
-      /*--- Normal vector for this vertex (negate for outward convention) ---*/
+    auto V_infty = GetCharacPrimVar(val_marker, iVertex);
 
-      geometry->vertex[val_marker][iVertex]->GetNormal(Normal);
-      for (iDim = 0; iDim < nDim; iDim++) Normal[iDim] = -Normal[iDim];
-      conv_numerics->SetNormal(Normal);
+    /*--- Index of the closest interior node ---*/
 
-      /*--- Retrieve solution at the farfield boundary node ---*/
+    Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
 
-      V_domain = nodes->GetPrimitive(iPoint);
+    /*--- Normal vector for this vertex (negate for outward convention) ---*/
 
-      /*--- Recompute and store the velocity in the primitive variable vector. ---*/
+    geometry->vertex[val_marker][iVertex]->GetNormal(Normal);
+    for (iDim = 0; iDim < nDim; iDim++) Normal[iDim] = -Normal[iDim];
+    conv_numerics->SetNormal(Normal);
 
-      for (iDim = 0; iDim < nDim; iDim++)
-        V_infty[iDim+1] = GetVelocity_Inf(iDim);
```

Comment:
These calls V_infty[iDim+1],  are one of the main differences between the NEMO solvers and everything.  Would it be wasteful to use the the NEMO indices through the code?  I haven't looked though enough of the code to know the full impact, but it seems that it could go a long way to reduce duplication.

Reply:
It would be a little yes, because the velocity offset for nemo is not known at compile-time.

But I have something that might interest you in a subsequent PR.

Reply:
Oh boy, I cant wait